### PR TITLE
adding ext schemas for customScriptForLinux, LinuxDiagnostic, and vma…

### PR DESCRIPTION
**Pull request by jluk**
…ccessForLinux

This is strictly for testing - please only enable in BrazilUS

### CODE REVIEW

File: schemas/2015-08-01/Microsoft.Compute.json [hunk 1]
```diff
@@ -1106,18 +1116,328 @@
...
+          "minLength": 1
+        },
+        "autoUpgradeMinorVersion": {
+          "type": "boolean"
+        },
+        "settings": {
+          "type": "object",
+          "properties": {
+            "fileUris": {
+              "type": "array"
```
Comment (TianoMS) [inline schemas/2015-08-01/Microsoft.Compute.json]: Specify the item type here?

File: schemas/2015-08-01/Microsoft.Compute.json [hunk 2]
```diff
@@ -1106,18 +1116,328 @@
...
+        "typeHandlerVersion",
+        "autoUpgradeMinorVersion",
+        "settings",
+        "protectedSettings"
+      ]
+    },
+    "bgInfo": {
+      "type": "object",
+      "properties": {
+          "publisher": {
```
Comment (TianoMS) [inline schemas/2015-08-01/Microsoft.Compute.json]: Please follow the indent convention. We use two spaces.

File: schemas/2015-08-01/Microsoft.Compute.json [hunk 3]
```diff
@@ -1106,18 +1116,328 @@
...
+              "type": "string"
+            },
+            "storageAccountName": {
+              "type": "string"
+            },
+            "storageAccountKey": {
+              "type": "string"
+            }
+          },
+          "required": [
```
Comment (TianoMS) [inline schemas/2015-08-01/Microsoft.Compute.json]: Please remove "required" property if no required properties.

File: schemas/2015-08-01/Microsoft.Compute.json [hunk 4]
```diff
@@ -192,7 +192,12 @@
             { "$ref": "#/definitions/genericExtension" },
             { "$ref": "#/definitions/iaaSDiagnostics" },
             { "$ref": "#/definitions/iaaSAntimalware" },
-            { "$ref": "#/definitions/customScriptExtension" }
+            { "$ref": "#/definitions/customScriptExtension" },
+            { "$ref": "#/definitions/customScriptForLinux"},
```
Comment (TianoMS) [inline schemas/2015-08-01/Microsoft.Compute.json]: Please add test for the change.

Comment (jluk) [inline schemas/2015-08-01/Microsoft.Compute.json]: sorry how should I test this change?

Comment (TianoMS) [inline schemas/2015-08-01/Microsoft.Compute.json]: Check our Repo home: https://github.com/Azure/azure-resource-manager-schemas. You can add one test case in azure-resource-manager-schemas/tests/2015-08-01/Microsoft.Compute.tests.json, or you can modify an existing one.

Comment (jluk) [inline schemas/2015-08-01/Microsoft.Compute.json]: What exactly are these tests doing? I've added one for each extension added, but the existing tests are using old apiversions that wouldn't support a GET?

Comment (TianoMS) [inline schemas/2015-08-01/Microsoft.Compute.json]: This should be good enough.

**Pull request merged**